pkg/logql/log/parser.go

```diff
@@ -486,3 +500,76 @@ func (u *UnpackParser) unpack(it *jsoniter.Iterator, entry []byte, lbs *LabelsBu
 	}
 	return entry, nil
 }
+
+type SyslogParser struct {
+	maxMessageLength int
+	lbs              *LabelsBuilder
+}
+
+func NewSyslogParser() *SyslogParser {
+	return &SyslogParser{maxMessageLength: 1 << 12}
+}
+
+func (l *SyslogParser) Process(line []byte, lbs *LabelsBuilder) ([]byte, bool) {
+	if lbs.ParserLabelHints().NoLabels() {
+		return line, true
+	}
+	l.lbs = lbs
+	b := line[0]
+	r := bytes.NewReader(line)
```

Comment:
This could be in the syslogparser struct, and reused using Reset.

Which make me think can you compare this parser with other parser within the parser benchmark I'm curious how it performs.

Reply:
> Which make me think can you compare this parser with other parser within the parser benchmark I'm curious how it performs.

Unfortunately, it performs very badly :(

```console
$ go test ./pkg/logql/log -test.bench=Benchmark_Parser -test.run=^$
goos: linux
goarch: amd64
pkg: github.com/grafana/loki/pkg/logql/log
cpu: 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
Benchmark_Parser/json/no_labels_hints-8         	  403036	      2986 ns/op
Benchmark_Parser/json/labels_hints-8            	  549129	      2337 ns/op
Benchmark_Parser/jsonParser-not_json_line/no_labels_hints-8         	51310922	        23.52 ns/op
Benchmark_Parser/jsonParser-not_json_line/labels_hints-8            	51239817	        24.83 ns/op
Benchmark_Parser/unpack/no_labels_hints-8                           	 1317918	       845.1 ns/op
Benchmark_Parser/unpack/labels_hints-8                              	 1346008	       889.3 ns/op
Benchmark_Parser/unpack-not_json_line/no_labels_hints-8             	48373522	        26.34 ns/op
Benchmark_Parser/unpack-not_json_line/labels_hints-8                	46241360	        26.00 ns/op
Benchmark_Parser/logfmt/no_labels_hints-8                           	  778330	      1506 ns/op
Benchmark_Parser/logfmt/labels_hints-8                              	  815367	      1513 ns/op
Benchmark_Parser/regex_greedy/no_labels_hints-8                     	  418903	      2964 ns/op
Benchmark_Parser/regex_greedy/labels_hints-8                        	  412575	      3015 ns/op
Benchmark_Parser/regex_status_digits/no_labels_hints-8              	 4998464	       236.7 ns/op
Benchmark_Parser/regex_status_digits/labels_hints-8                 	 5097273	       234.4 ns/op
Benchmark_Parser/pattern/no_labels_hints-8                          	 8696990	       133.9 ns/op
Benchmark_Parser/pattern/labels_hints-8                             	10077829	       115.7 ns/op
Benchmark_Parser/syslog/no_labels_hints-8                           	  399288	      2853 ns/op
Benchmark_Parser/syslog/labels_hints-8                              	  420158	      2858 ns/op
PASS
ok  	github.com/grafana/loki/pkg/logql/log	24.625s
```

Reply:
Do you know why :) ?